internal/reader/ccip.go

```diff
@@ -57,6 +58,13 @@ type CCIP interface {
 	// TODO: if destination was a parameter, this could be a capability reused across plugin instances.
 	NextSeqNum(ctx context.Context, chains []cciptypes.ChainSelector) (seqNum []cciptypes.SeqNum, err error)
 
+	// Nonces fetches all nonces for the provided selector/address pairs. Addresses are the hex encoded raw address.
```

Comment:
Hmm the hex encoding here may be problematic since thats just an EVM thing. Other chains use e.g base58

Reply:
This should be fine because it's only used as a UID. Internally address are stored as `cciptypes.Bytes` and we would use the `String()` function for a string representation of those bytes. Chain specific encoding is not important until we start building merkle roots, that happens elsewhere.